### PR TITLE
Moving logic for skipping hosts in pr cases

### PR DESCRIPTION
**Pull request by TerrenceMcGuinness-NOAA**
# Description

This PR removes the logic of skipping hosts for pr cases from `create_experiment.py` and moves it to a test in the cron bash driver using a `parse_yaml.py` python tool.

The Jenkins pipeline was not effected as it uses the `get_host_case_list.py` utility to form the cases on a per host bases.

# Type of change

- Maintenance (simple CI code refactor)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?
Being tested in CI BASH dev space on Orion to see if cases are still being skipped and reported as such.
**NOTE:** The Fail is a success in this test as it is the gefs case and does not have an alternative build in bash CI.

### CODE REVIEW

File: ci/scripts/driver.sh [hunk 1]
```diff
@@ -220,18 +220,20 @@ for pr in ${pr_list}; do
       set +e
       export LOGFILE_PATH="${HOMEgfs}/ci/scripts/create_experiment.log"
       rm -f "${LOGFILE_PATH}"
+      yaml_case_file="${HOMEgfs}/ci/cases/pr/${case}.yaml"
+      skip_hosts=$("${HOMEgfs}/ci/scripts/utils/pars_yaml.py" --yaml "${yaml_case_file}" --key skip_ci_on_hosts --string)
```
Comment (aerorahul) [inline ci/scripts/driver.sh]: I think there is a typo in the name of this script pars***e***_yaml.py

File: ci/scripts/utils/pars_yaml.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python3
```
Comment (aerorahul) [inline ci/scripts/utils/pars_yaml.py]: Consider fixing the typo in the name of this script.
pars***e***_yaml.py

File: ci/scripts/utils/pars_yaml.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    Args:
+        yamlfile (Path): The path to the yaml file.
+        key (str): The key to return the value of.
+
+    Returns:
+        The value of the specified key in the yaml file.
+    """
+
+    data = AttrDict(HOMEgfs=_top)
+    data.update({'HOMEgfs': _top})
```
Comment (aerorahul) [inline ci/scripts/utils/pars_yaml.py]: Is the `update` required if AttrDict is creating a dict with `HOMEgfs` already with the right value `_top`?

**Review by aerorahul — COMMENTED** [review]
looks good.
just a typo in the name of the script and a q.

File: ci/scripts/utils/parse_yaml.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    Parse command-line arguments.
+
+    Returns:
+        argparse.Namespace: The parsed command-line arguments.
+    """
+
+    parser = ArgumentParser(description=description)
+    parser.add_argument('-y', '--yaml', help='full path to yaml file to parce', type=Path, required=True)
+    parser.add_argument('-k', '--key', help='key to return value of', type=str, required=True)
+    parser.add_argument('-s', '--string', help='output results as stings', action="store_true", required=False)
```
Comment (aerorahul) [inline ci/scripts/utils/parse_yaml.py]:
```suggestion
    parser.add_argument('-s', '--string', help='output results as strings', action="store_true", required=False)
```

File: ci/scripts/utils/parse_yaml.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+def parse_args():
+    """
+    Parse command-line arguments.
+
+    Returns:
+        argparse.Namespace: The parsed command-line arguments.
+    """
+
+    parser = ArgumentParser(description=description)
+    parser.add_argument('-y', '--yaml', help='full path to yaml file to parce', type=Path, required=True)
```
Comment (aerorahul) [inline ci/scripts/utils/parse_yaml.py]:
```suggestion
    parser.add_argument('-y', '--yaml', help='full path to yaml file to parse', type=Path, required=True)
```

**Review by aerorahul — APPROVED** [review]

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Comment by WalterKolczynski-NOAA** [comment]
@TerrenceMcGuinness-NOAA can we merge this?

**Pull request merged**